\documentclass[11pt,a4paper]{article}

\usepackage[margin=2cm]{geometry}
\usepackage[utf8]{inputenc}
\usepackage[english]{babel}
\usepackage{amsmath}
\usepackage{amsfonts}
\usepackage{amssymb}
\usepackage{graphicx}
\usepackage{hyperref}

%added userpackages
\usepackage{microtype}
\usepackage{natbib}
\usepackage{lmodern}
\usepackage{bmpsize-base}



%%%%%%%%%%%%%%%%%%%%%%%%%

\title{Graph-Based Analysis of Gene Activity in Lung Cancer based on Protein-Interaction Networks}
\author{
Simone Bergmann, 3957624 \\
\texttt{simone.bergmann@uni-bielefeld.de}%
}
%%%%%%%%%%%%%%%%%%%%%%%%%
\twocolumn
\begin{document}
\maketitle

% FACTS:
% 15 bis 30 Seiten (exklusive Titel, Abstract, Inhaltsverzeichnis, Literaturverzeichnis, ...)
% Deckblatt
% Inhaltsverzeichnis
% Nicht zu viele Unterkapitel / zu sehr verschachtelt
% ca 30 Quellen
% Jedes Kapitel am Anfang und am Ende zusammenfassen aber Fokus auf Ende

\cite{page1999pagerank}

Task:
Creation of a data model and integration of genes and proteins with interactions in an appropriate form within a graph database,
which is to be used for the identification of relevant genes for distinguishing between healthy tissue and lung cancer.

Key facts:
* cancerous mean tpm from Cell Model Passport
* healthy mean tpm from Genotype-Tissue Expression
* delta tpm as difference between healthy and cancerous tpm
* genes are connected with proteins and proteins are connected with each other (STRING database)
* neo4J as graph database
* pagerank as graph algorithm to identify relevant genes

% TODO
% Abstact soll kursiv sein

\input{capitle/00_abstract}

\input{capitle/01_0_intro}

\input{capitle/02_0_background}

\input{capitle/03_experimental_setup}

\input{capitle/04_results}

\input{capitle/05_0_discussion}

\input{capitle/06_conclusion}

Content:

preparation of the Datasets for the graph db
* for every dataset we need the format genes as rows and at least the ENS ID and a mean tpm for every gene row
    * GTEX (Genotype-Tissue Expression) - healthy tissues
        * downloaded bulk-tissue-expression file (... more about the data)
        * format was: Gene / ENS ID as rows and tissues as columns - tpm value for every gene in every tissue
        * melted down to genes as rows and tissue and tpm value as columns
        * grouped by tissues to get a single tpm value for every gene (eigentlich sum and count to get mean)
    → 56.156 human genes with mean values for healthy tpm

    * CMP (Cell Model Passport) - cancerous tissues
        * downloaded all RNA-Seq processed data - which contain data from Sanger Institute and from Broad Institute
        * format was: genes with CMP IDs as rows and tpms, tissues (called model), ... per column
        * separate file loaded with model annotation to get a list of all models (tissues) that have lung cancer as cancer type
        * filtered the data to get only the lung cancer models
        * grouped all models (tissues) to get a mean tpm value for every gene
        * needed to add ENS ID to the genes for further processing
        * separate ensemble file downloaded from biomart (https://www.ensembl.org/biomart/martview/)
        * contains Gene stable ID and gene symbol (name)
        * idea is to add the Gene stable ID by merging both tables by the gene name
        * daher gehen wir davon aus, dass jeder gen name eindeutig ist
        * war es nicht - 10.605/48.311 rows in the ensemble file had a not unique gene name
        * rows with names that are not unique were removed
        * merging the ensemble file with the CMP data to get the ENS ID for every gene
        * 3.760 / 37.262 genes had no ENS ID → missing IDs could be added by synonymes maybe because of duplicate removement
        * these genes are removed
    → 33.502 human genes with mean values tpm for lung cancer

    * HPA (Human Protein Atlas) - cancerous tissues
        * downloaded RNA HPA cell line cancer gene data from the HPA website
        * format was: genes with ENS ID as rows and Gene Names, Cancertype and TPM value as columns
        * data filtered for lung cancer
    → 20.162 human genes with tpm values for lung cancer
    → DELETED!!!

Datasets to nodes and edges
    * genes as nodes
        * eigenschaft: name, gene stable id, norm healyth tpm, norm cancerous tpm, delta tpm, z score, delta tpm relevant
        * merging the GTEx (healthy) and CMP (cancerous) data to get a list of all genes that are in both datasets
        * normalizing the TPM Values with log scaling
        * calculate a delta TPM value for every gene as the difference between the mean tpm value in the healthy and the cancerous dataset
        * NOT THE ABSOLUTE
        * to find out which genes have a relevant change in the healthy and cancerous tpm value we calculate the z score for the delta tpm values for every gene
        * z score is calculated as the difference between the delta tpm value and the mean delta tpm value divided by the standard deviation of the delta tpm values
        * z score is a measure of how many standard deviations an element is from the mean
        * 1.643(?!) genes with a z score of 1,96 or higher are considered as delta tpm relevant (p Wert damit 0,05 and Konfidenzniveau 95%)
    → 32.865 gene Nodes

    * protein gene connection as edges
        * eigenschaft: gene stable id, protein stable id
        * proteins are needed as a second node type to get a connection between the genes.
        * downloaded protein file from biomart with gene stable ID and protein stable ID
        * filtered for those gene stable IDs that are in the gene node list.
        * we only have those proteins that have a connection (was sagt die tabelle genau aus?)
    → 101.731 protein gene edges (means: every protein is connected to exactly one gene, but every gene can have multiple proteins)

    * proteins as nodes
        * eigenschaft: protein stable id
        * use the protein gene table as base
        * only use the column with the protein stable ID
    → 101.731 protein Nodes

    * protein protein connection as edges
        * eigenschaft: protein stable id 1, protein stable id 2
        * downloaded protein protein interaction file from STRING database
        * filtered for those protein stable IDs that are in the protein node list
    → 11.247.242 protein protein edges
    (not every protein is connected to another one → but since every gene could have multiple proteins, the connected gene may have a connection to another protein)
    TODO: are those genes that are not connected to another gene relevant for the analysis?

Graph database
    * Neo4J
    * gene nodes query:
    * protein nodes query:
    * gene protein edges query:
    * protein protein edges query:




ANALYSIS
IN ONENOTE


\newpage
\bibliographystyle{plain} % We choose the "plain" reference style
\bibliography{refs} % Entries are in the refs.bib file

\end{document}